### PR TITLE
feat: add watsonx ai provider

### PR DESCRIPTION
**Pull request by liyanwei93**
Signed-off-by: yanweili <yanweili@ibm.com>

Closes # 

## 📑 Description

Add IBM watsonx ai as AI provider for k8sGPT.
Document: https://github.com/k8sgpt-ai/docs/pull/107

PR in k8sgpt: https://github.com/k8sgpt-ai/k8sgpt/pull/1163

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by gyliu513** [comment]
This has a dependency with https://github.com/k8sgpt-ai/k8sgpt/pull/1234

**Comment by gyliu513** [comment]
@liyanwei93 can you help update this PR a bit based on https://github.com/k8sgpt-ai/k8sgpt/pull/1234? I have renamed `watsonxai` to `ibmwatsonxai`, thanks!

File: api/v1alpha1/k8sgpt_types.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ const (
 	Cohere          = "cohere"
 	Google          = "google"
 	GoogleVertexAI  = "googlevertexai"
-	WatsonXAI       = "watsonxai"
+	IBMWatsonXAI    = "ibmwatsonxai"
```
Comment (gyliu513) [inline api/v1alpha1/k8sgpt_types.go:143]: I prefer we use `IBMWatsonxAI`

Comment (liyanwei93) [reply api/v1alpha1/k8sgpt_types.go]: Updated

**Comment by gyliu513** [comment]
hey @AlexsJones can you help review or delegate to others for review? Thanks

**Comment by liyanwei93** [comment]
Hi @AlexsJones, all dependent PRs have been merged. Could you please help to review this one. Thanks in advanced.

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**